TF comparison updates (#985)
pulumi/pulumi-hugo: themes/default/content/docs/intro/vs/terraform/_index.md

> ---

Author: This page shows net-new because it was moved from above, to make room for the sibling "equivalents" page.

title: Pulumi vs. Terraform

> meta_desc: "Pulumi and Terraform have a few similarities, but they differ in many key ways. This page helps provide a rundown of the differences."

Author: Changed this meta description to be a bit more page-specific (using the lead sentence from the current page). Open to suggestions if folks have better ideas of course!

| [Testing and Validation](#testing) | Unit, property, and integration testing. Supports popular test frameworks. | Integration testing only |
| [Modes of Execution](#modes) | Run CLI commands or initiate commands programmatically with Automation API. | Run CLI commands or perform remote runs with SaaS offering. |
| [Embed within Application Code](#embedding) | Yes, via Automation API | No |
| [Third-party CI/CD Tools Support](#cicd) | Yes | Yes |
| [Policy as Code](#policy) | Yes | Yes |
| [Secrets Management](#secrets) | Yes. Secrets are encrypted in transit and in the state file. | No. Secrets are stored in a  separate product (Vault). There is no way to encrypt them in the state file. |
| [Audit Capabilities](#auditing) | Yes | Limited |
| [Adopt Existing Resources](#adopting) | Yes. Generates code as part of the import process. | Yes. No code generation capabilities. |
| [Aliases](#aliases) | Yes | Limited |
| [Transformations](#transformations) | Yes | No |
| [Import Code from other IaC Tools](#converting) | Yes | No |

> Getting started with Pulumi is easy. If you have experience with Terraform and already have HCL, you can convert it to Pulumi. Follow our [Converting Terraform HCL to Pulumi]({{< relref "/docs/guides/adopting/from_terraform#converting-terraform-hcl-to-pulumi" >}}) guide. If you would like a reference on Terraform terminology or command equivalents in Pulumi, review our [Terminology Equivalence]({{< relref "/docs/intro/vs/terraform/terminology" >}}) table. If you would like to deploy a simple program, follow our Get Started guide:

Author: ```suggestion
Getting started with Pulumi is easy. If you have experience with Terraform and already have HCL, you can convert it to Pulumi. Follow our [Converting Terraform HCL to Pulumi]({{< relref "/docs/guides/adopting/from_terraform#converting-terraform-hcl-to-pulumi" >}}) guide. If you would like a reference comparing Terraform terminology or commands to their Pulumi equivalents, see [Terminology and Command Equivalence]({{< relref "/docs/intro/vs/terraform/terminology" >}}). If you would like to deploy a simple program, follow our Get Started guide:
```

> Getting started with Pulumi is easy. If you have experience with Terraform and already have HCL, you can convert it to Pulumi. Follow our [Converting Terraform HCL to Pulumi]({{< relref "/docs/guides/adopting/from_terraform#converting-terraform-hcl-to-pulumi" >}}) guide. If you would like a reference on Terraform terminology or command equivalents in Pulumi, review our [Terminology Equivalence]({{< relref "/docs/intro/vs/terraform/terminology" >}}) table. If you would like to deploy a simple program, follow our Get Started guide:

Author: ```suggestion
Getting started with Pulumi is easy. If you have experience with Terraform and already have HCL, you can convert it to Pulumi. Follow our [Converting Terraform HCL to Pulumi]({{< relref "/docs/guides/adopting/from_terraform#converting-terraform-hcl-to-pulumi" >}}) guide. If you would like a reference on Terraform terminology or command equivalents in Pulumi, see [Terminology and Command Equivalence]({{< relref "/docs/intro/vs/terraform/terminology" >}}) table. If you would like to deploy a simple program, follow our Get Started guide:
```
